### PR TITLE
Ensure net7.0-rc2 support

### PR DESCRIPTION
**Pull request by g7ed6e**

### CODE REVIEW

**Comment by pushack** [comment]
I have got the same problems testing against net 7.0. Some types accessed by reflection in DataContractEx class have changed namespaces in net 7.0. For example line 89 in DataContractEx.cs:
     
internal static Type DataContractType => typeof(DataContractSerializer).Assembly.GetType("System.Runtime.Serialization.DataContract");

The DataContract class has been moved to the System.Runtime.Serialization.DataContracts namespace. So the change must be compatible for both variants:

internal static Type DataContractType => GetDataContractType();

    private static Type GetDataContractType()
    {
        var dataContractType = typeof(DataContractSerializer).Assembly.GetType("System.Runtime.Serialization.DataContract");
        dataContractType ??= typeof(DataContractSerializer).Assembly.GetType("System.Runtime.Serialization.DataContracts.DataContract");
        if (dataContractType is null)
        {
            throw new NullReferenceException("DataContract Type not found");
        }
        return dataContractType;
    }

**Comment by g7ed6e** [comment]
> I have got the same problems testing against net 7.0. Some types accessed by reflection in DataContractEx class have changed namespaces in net 7.0. For example line 89 in DataContractEx.cs:
> 
> internal static Type DataContractType => typeof(DataContractSerializer).Assembly.GetType("System.Runtime.Serialization.DataContract");
> 
> The DataContract class has been moved to the System.Runtime.Serialization.DataContracts namespace. So the change must be compatible for both variants:
> 
> internal static Type DataContractType => GetDataContractType();
> 
> ```
> private static Type GetDataContractType()
> {
>     var dataContractType = typeof(DataContractSerializer).Assembly.GetType("System.Runtime.Serialization.DataContract");
>     dataContractType ??= typeof(DataContractSerializer).Assembly.GetType("System.Runtime.Serialization.DataContracts.DataContract");
>     if (dataContractType is null)
>     {
>         throw new NullReferenceException("DataContract Type not found");
>     }
>     return dataContractType;
> }
> ```

Thanks for the hint!

**Comment by mconnew** [comment]
I don't think this is the correct approach. The reason for all the reflection code is that XsdDataContractExporter had some major bugs in it which made it completely unusable. Those should all be fixed now, so we should be able to replace usage of my wrapping class XsdDataContractExporterEx with the correct XsdDataContractExporter class. It's possible there's some lingering bugs, but I think it should all be fixed now. Can you try that approach instead of modifying the wrapper classes?

**Comment by g7ed6e** [comment]
@mconnew All unit tests are successful with XsdDataContractExporter when targeting  .NET 7 and .NET 472. However we still need XsdDataContractExporterEx for .NET6/5/3.1.
I think the fixed .NET7 implementation is also available in https://www.nuget.org/packages/System.Runtime.Serialization.Schema/7.0.0-rc.1.22426.10 but this package does not target netstandard2.0.

I introduced a runtime check to use XsdDataContractExporter only when running .NET7 and fallback to existing XsdDataContractExporterEx.

I’m also wondering what should we do regarding the AppContext switch introduced a few time ago.

**Comment by mconnew** [comment]
The new Schema package can't target netstandard2.0 as it depends on some types which were internal to Datacontractserializer on previous versions of .NET and have been moved to a new namespace and made public in 7, as well as having some of the member names modified. The sticking point was that the code depends on CodeDom (which can't be part of .NET runtime) and this was the best of multiple less than ideal options to solve the problem. 
Runtime switching between the two implementations is the right way to do it. We can drop the Ex implementation in 2 years when .NET 6 goes EOL.

**Review by mconnew — APPROVED** [review]

File: src/CoreWCF.NetTcp/tests/CoreWCF.NetTcp.Tests.csproj [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="2.1.3" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1' OR '$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0'">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1' OR '$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0'">
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
[the remark this answers is not in the document]
Comment (g7ed6e) [reply src/CoreWCF.NetTcp/tests/CoreWCF.NetTcp.Tests.csproj:34]: I inversed the condition. Thanks!

**Pull request merged**